packages/components/src/components/Pagination/usePagination.tsx

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from "react";
+
+const DOTS = "...";
+
+const range = (start: number, end: number): string[] => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => `${idx + start}`);
+};
+
+export const usePagination = (
```

Comment:
I think here it would be actually nice to have a short comment describing what this hook does on the high level.